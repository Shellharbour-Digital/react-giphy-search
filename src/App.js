import React, { Component } from 'react';

import { SearchBox } from './components/search-box/search-box.component';
import GiphyList from './components/giphy-list/giphy-list.component';

class App extends Component {
  constructor() {
    super();

    this.state = {
      url: 'https://api.giphy.com/v1/gifs/trending?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&limit=50&rating=g'
    };
  }

  handleChange = e => {
    e.target.value === '' ? (
      this.setState({
        url: `https://api.giphy.com/v1/gifs/trending?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&limit=50&rating=g`
      })
    ) : (
      this.setState({
        url: `https://api.giphy.com/v1/gifs/search?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&q=${e.target.value}&limit=50&offset=0&rating=g&lang=en`
      })
    )
  }

  render() {
    const { url } = this.state;
    return (
      <div className='App'>
        <SearchBox
          placeholder='ðŸ”Ž search gifs'
          handleChange={this.handleChange}
          debounceTimeout={300}
        />
        <GiphyList
          dataSource={url}
        />
      </div>
    );
  }
}

export default App;
