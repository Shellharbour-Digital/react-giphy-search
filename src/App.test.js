import { shallow } from 'enzyme';
import React, { Component } from 'react';
import App from './App';

describe('Spinner component', () => {
  let wrapper = shallow(<App />);

  it('expect to render App', ()  => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to handleChange', () => {
    wrapper.instance().handleChange({ target: { value: 'test' }});
    expect(wrapper.state().url).toEqual(`https://api.giphy.com/v1/gifs/search?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&q=test&limit=50&offset=0&rating=g&lang=en`);
    wrapper.instance().handleChange({ target: { value: '' }});
    expect(wrapper.state().url).toEqual(`https://api.giphy.com/v1/gifs/trending?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&limit=50&rating=g`);
  });
});
