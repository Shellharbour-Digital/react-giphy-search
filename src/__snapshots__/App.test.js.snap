// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component expect to render App 1`] = `
<div
  className="App"
>
  <SearchBox
    debounceTimeout={300}
    handleChange={[Function]}
    placeholder="ðŸ”Ž search gifs"
  />
  <WithGifs
    dataSource="https://api.giphy.com/v1/gifs/trending?api_key=AscSI3FnS69DmOLqAbNa4GZZQvEeNWil&limit=50&rating=g"
  />
</div>
`;
